Userland/Libraries/LibJS/Heap/Heap.cpp

```diff
@@ -266,7 +267,9 @@ void Heap::collect_garbage(CollectionType collection_type, bool print_report)
 
 void Heap::gather_roots(HashMap<Cell*, HeapRoot>& roots)
 {
-    vm().gather_roots(roots);
+    if (m_gather_roots)
+        m_gather_roots(roots);
```

Comment:
This name is a bit confusing when the method name is also `gather_roots`. Perhaps, "m_gather_embedder_roots" or similar?